### PR TITLE
Awesome books refactor to use javascript classes

### PR DESCRIPTION
**Pull request by pulgosodog**
Refactor previous code acording to **Awesome books: refactor to use JavaScript classes** activity:

Add a object `constructor`.
Added `functions `inside the class to be `methods`.
Rework `functions` to work as `methods`.
Refactor events to work with `class ` `methods` without affecting functionality nor adding bugs.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,64 +1,70 @@
...
 
-function addBook(tittleName, authorName) {
-  bookLibrary[tittleName] = {
-    tittle: tittleName,
-    author: authorName,
-  };
-  const jsonBooks = JSON.stringify(bookLibrary);
-  localStorage.setItem('books', jsonBooks);
-}
+let jsonBooks = JSON.parse(localStorage.getItem('booksStorage'));
```
Comment (emmiiorji) [inline app.js]: When I load the page for the first time, the `jsonBooks` variable is undefined and so I get an error resulting from an attempt to call the map function. To fix the issue, you can initialize an empty array when the key wasn't found in the local storage.
e.g
```javascript
let jsonBooks = JSON.parse(localStorage.getItem('booksStorage')) || [];
```
image

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @pulgosodog, 
# Status: Changes Required ♻️

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<div align='center'>

</div>

## Good Highlights✔️

- Solution has been refactored using JavaScript classes
- Given wireframe is maintained


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.js (hunk 1 above)
Comment (pulgosodog) [reply app.js]: Fixed! I moved `jsonBooks = JSON.parse(localStorage.getItem('booksStorage'));` and `bookLibrary = jsonBooks.map((book) => new Books(book.title, book.author));` to the `DOMContentLoaded` event, and set the `jsonBooks` when it is declared to an emty array.

Comment (pulgosodog) [reply app.js]: Appreciate the feedback! 🙌

**Review by franclobo — APPROVED** [review]
Hi @pulgosodog, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Correct Github flow ✔️ 
- Descriptive PR ✔️ 
- Correct use of JavaScript classes ✔️ 
- Good communication with CR ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**